File: platform/common-lib/src/main/java/com/researchex/common/messaging/AvroSerializationException.java
package com.researchex.common.messaging;

/** Avro 직렬화/역직렬화 과정에서 발생한 예외를 표현한다. */
public class AvroSerializationException extends RuntimeException {

  public AvroSerializationException(String message) {
    super(message);
  }

  public AvroSerializationException(String message, Throwable cause) {
    super(message, cause);
  }
}

File: platform/common-lib/src/main/kotlin/com/researchex/common/messaging/AvroSerializationException.kt
package com.researchex.common.messaging

/** Avro 직렬화/역직렬화 과정에서 발생한 예외를 표현한다. */
class AvroSerializationException : RuntimeException {
    constructor(message: String) : super(message)
    constructor(message: String, cause: Throwable) : super(message, cause)
}
